hantec_api_ecommerce/controllers/main.py

```diff
@@ -0,0 +1,570 @@
+
+    @route('/register_payment_invoice/<model("account.move"):invoice>', methods=['POST'], type='json', auth='user')
+    def register_payment(self, invoice=False):
+        data = request.jsonrequest
+
+        amount = data.get('amount')
+        journal_id = data.get('journal_id')
+        payment_method_id = data.get('payment_method_id')
+
+        register_payment_wizard = request.env['account.payment.register'].with_context(
+            active_model='account.move',
+            active_ids=[invoice.id]
+        )
+
+        wizard = register_payment_wizard.create({
+            'amount': amount,
+            'journal_id': journal_id,
+            'l10n_mx_edi_payment_method_id': payment_method_id,
+            'communication': 'Payment for invoice %s' % invoice.name,
+        })
+
+        # Registrar el pago
+        wizard.action_create_payments()
+
+        return {'success': 'El pago ha sido registrado correctamente.'}
+    
+    @route('/get_shipping_info/<model("sale.order"):order>', methods=['GET'], type='json', auth='user')
+    def get_shipping_info(self, order=False):
+
+        # Datos de la dirección de entrega
+        shipping_address = order.partner_shipping_id
+        address_data = {
+            'name': shipping_address.name,
+            'phone': shipping_address.phone,
+            'email': shipping_address.email,
+            'street': shipping_address.street,
+            'street2': shipping_address.street2,
+            'city': shipping_address.city,
+            'state': shipping_address.state_id.name if shipping_address.state_id.name else '',
+            'zip': shipping_address.zip,
+            'country': shipping_address.country_id.name if shipping_address.country_id else ''
+        }
+
+        # Obtener los registros de envío relacionados
+        pickings = order.picking_ids
+
+        # Preparar datos de los envíos para la respuesta
+        shipping_data = []
+        for picking in pickings:
+            lines_data = [{
+                'product': line.product_id.name,
+                'quantity': line.product_uom_qty,
+                'done': line.quantity_done,
+                'name': line.name
+            } for line in picking.move_lines]
+
+            shipping_info = {
+                'name': picking.name,
+                'scheduled_date': picking.scheduled_date,
+                'state': picking.state,
+                'carrier': picking.carrier_id.name if picking.carrier_id else 'No definido',
+                'tracking_reference': picking.carrier_tracking_ref,
+                'lines': lines_data,
+                'address_data': address_data
+            }
+            shipping_data.append(shipping_info)   
+
+        shipping_data.append({
+            "sales_team": order.team_id.id,
+            "market_place_reference": order.channel_order_reference,
+            "sale_order_name": order.name,
+            "picking_id": picking.id,
```

Comment:
Using possibly undefined loop variable 'picking'

picking is defined as a for loop var only